### PR TITLE
feat: add spark3

### LINKED ISSUE #256: Add Spark 3 roles and playbooks
Spark 3 comes with a lot of new features that make it very important to support it in TDP.
> Here are the feature highlights in Spark 3.0: adaptive query execution; dynamic partition pruning; ANSI SQL compliance; significant improvements in pandas APIs; new UI for structured streaming; up to 40x speedups for calling R user-defined functions; accelerator-aware scheduler; and SQL reference documentation.

The approach chosen is to **install Spark 3 alongside Spark 2** in a similar fashion to how Python 3 is installed alongside Python2. This is also the way things are done in CDP with [CDS](https://docs.cloudera.com/cdp-private-cloud-base/7.1.6/cds-3/topics/spark-running-spark3-apps.html).

 This means a totally isolated install:
- New Ansible groups: `spark3_hs`, `spark3_client`
- A dedicated install dir: `/opt/tdp/spark3`; and distinct CLIs: `spark3-submit`, `spark3-shell`, `spark3-sql`, `pyspark3`
- A new service `spark3-history-server`

The Spark version is 3.2 based on the branch [branch-3.2-TDP](https://github.com/TOSIT-IO/spark/tree/branch-3.2-TDP) that requires Python 3.6+ and comes with enhanced support for pandas. Therefore the following packages will have to be installed on all `yarn_nm` hosts:
- OS packages: `python3`, `python3-pip`, `python3-setuptools`
- PyPI packages: `numpy~=1.19`, `pandas~=1.0`, `pyarrow~=4.0`

### PR DESCRIPTION
**Pull request by Nuttymoon**
#### Which issue(s) this PR fixes:

Fixes #256 

#### Additional comments:

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) and that you give permission to [TOSIT](https://www.tosit.io/),
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: roles/spark/common/tasks/python3_redhat.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Copyright 2022 TOSIT.IO
+# SPDX-License-Identifier: Apache-2.0
+
+---
+- name: Install python3
+  yum:
```
Comment (rpignolet) [inline roles/spark/common/tasks/python3_redhat.yml:6]: We should not install package inside `tdp-collection`. If these packages are required, you should mention it inside the README for Spark3 roles.

We have some package installation inside `tdp-collection` but they will eventually be removed.

File: roles/spark/common/tasks/spark3_python3.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+- name: Upgrade pip3
+  pip:
+    name:
+      - pip
+    state: latest
+    executable: pip3
+
+- name: Install numpy, pandas and pyarrow
+  pip:
```
Comment (rpignolet) [inline roles/spark/common/tasks/spark3_python3.yml:23]: Same as packages, `pip` dependencies should not be installed by `tdp-collection`.

**Comment by rpignolet** [comment]
Commit messages does not respect the format we use.

**Comment by Nuttymoon** [comment]
Can I find a doc about the commit message format to use?

File: roles/spark/common/tasks/python3_redhat.yml (hunk 1 above)
Comment (Nuttymoon) [reply roles/spark/common/tasks/python3_redhat.yml:6]: Noted. I will move the packages install to `tdp-getting-started` and include them as requirements

**Comment by Nuttymoon** [comment]
@rpignolet I have removed Python 3 and PyPI packages install + renamed the commits

**Review by mehdibn — APPROVED** [review]

**Comment by mehdibn** [comment]
@Nuttymoon: now, we have to integrate spark3-hs to knox :)

**Review by leopaul36 — APPROVED** [review]
Good job @Nuttymoon !

**Review by gboutry — APPROVED** [review]
Deployed using `tdp-lib` and ran somes tests on it, worked well !

**Comment by Nuttymoon** [comment]
> @Nuttymoon: now, we have to integrate spark3-hs to knox :)

I think I did it here [knox.yml:119-121](https://github.com/TOSIT-IO/tdp-collection/pull/259/files#diff-e6fcb3ad1931c6905a8a6a18d7d0d3417ef07981ea41706675cedca919c5afceR119-R121). Isn't this enough?

**Comment by gboutry** [comment]
> > @Nuttymoon: now, we have to integrate spark3-hs to knox :)
> 
> @mehdibn I think I did it here [knox.yml:119-121](https://github.com/TOSIT-IO/tdp-collection/pull/259/files#diff-e6fcb3ad1931c6905a8a6a18d7d0d3417ef07981ea41706675cedca919c5afceR119-R121). Isn't this enough?

I think we should remove this. I deployed a spark3 without this property, and it worked properly through Knox, I think it's needed only for spark2 hs.

What's @mehdibn is implying is working on knox to ensure proper integration. From what I tested, almost everything works, except downloading `stdout` `stderr` logs because it redirects to YARNUIv2, which we do not deploy.

**Pull request merged**